BEIMA.Backend/MongoService/MongoConnector.cs

```diff
@@ -0,0 +1,192 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using MongoDB.Driver;
+using MongoDB.Bson;
+
+namespace BEIMA.Backend.MongoService
+{
+    public sealed class MongoConnector
+    {
+        //Public class members
+        public ServerType CurrentServerType { get; }
+
+        //Contains instance variables
+        private readonly MongoClient client = null;
+        private static readonly Lazy<MongoConnector> instance = new(() => new MongoConnector());
+
+        //Environment variables
+        private readonly string dbName = Environment.GetEnvironmentVariable("DatabaseName");
+        private readonly string deviceCollection = Environment.GetEnvironmentVariable("DeviceCollectionName");
+
+        //Singleton design pattern, used to get an instance of the MongoConnector
+        public static MongoConnector Instance { get { return instance.Value; } }
+
+        //Private constructor, used for singleton pattern. Cannot be called externally.
+        private MongoConnector()
+        {
+            string credentials;
+
+            if (Environment.GetEnvironmentVariable("CurrentEnv") == "dev-local")
+            {
+                CurrentServerType = ServerType.Local;
+                credentials = Environment.GetEnvironmentVariable("LocalMongoConnection");
+            }
+            else
+            {
+                CurrentServerType = ServerType.Cloud;
+                credentials = Environment.GetEnvironmentVariable("AzureCosmosConnection");
+            }
+            client = new MongoClient(credentials);
+        }
+
+        /*
```

Comment:
I have been writing my method/class headers using a different format, we should likely try to come up with a standard format. I think the one that I use is pretty nice because you only need to type '///' in VS and it will automatically put in a template for you.

Reply:
I think this is a good point to bring up in retro. Being consistent throughout our code base is important to the maintainability of BEIMA

Reply:
@josephmoore99 @AshlynAdamson should this be changed now, or should it wait until retro?

Reply:
If you agree with the header change, then we can do it now. But if you like what you had, then I think leaving it until retro is fine, gives us a chance to discuss about it

Reply:
Fixed c07b290

Reply:
I think the '///' is the standard for C#. Plus it also helps provide method info when hovering over a method call or class reference in VS.